coldfront/core/user/admin.py

```diff
@@ -17,3 +19,38 @@ def first_name(self, obj):
 
     def last_name(self, obj):
         return obj.user.last_name
+
+
+@admin.register(EmailAddress)
+class EmailAddressAdmin(admin.ModelAdmin):
+    list_display = ('user', 'email', 'is_primary', 'is_verified', )
+    ordering = ('user', '-is_primary', '-is_verified', 'email', )
+    list_filter = ('is_primary', 'is_verified', )
+    search_fields = ('user', 'email', )
+    fields = ('email', 'user', 'is_primary', 'is_verified', )
+    actions = ('make_primary', )
+    readonly_fields = ('is_primary', 'is_verified', )
+
+    # custom logic for delete button in emailaddress admin form
+    def delete_model(self, request, obj):
+        if obj.is_primary:
+            raise ValidationError('Cannot delete primary email. Unset primary status in list display before deleting.')
+        else:
+            super().delete_model(request, obj)
+
+    # custom action to give an email primary status without causing conflicting states
+    @admin.action(description='Make selected primary email')
+    def make_primary(self, request, queryset):
+        # (a) unset the current primary, (b) set the selected one as primary, and (c) update user.email
+        for emailaddress in queryset:
+            for primary_email in EmailAddress.objects.filter(user=emailaddress.user).filter(is_primary=True):
```

Comment:
FYI:
- You can place multiple filter clauses together:
    - `EmailAddress.objects.filter(user=emailaddress.user, is_primary=True)`.

Another FYI (feel free to ignore):
- Whether intentionally or not, it's good that you didn't call the [update](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#update) method directly on the filter results, because it "does not call any save() methods".
    - I.e., the custom `EmailAddress.save` logic you defined below wouldn't be called if you had used `update`.